components/PaymentMethodSelect.js

```diff
@@ -1,31 +1,48 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { injectIntl } from 'react-intl';
 
-import StyledSelect from './DeprecatedStyledSelect';
+import StyledSelect from './StyledSelect';
 import { paymentMethodLabelWithIcon } from '../lib/payment_method_label';
 
-const PaymentMethodSelect = ({ intl, paymentMethods, defaultPaymentMethod, ...props }) => {
-  return (
-    <StyledSelect
-      name="paymentMethod"
-      options={paymentMethods}
-      keyGetter="id"
-      defaultValue={defaultPaymentMethod}
-      {...props}
-    >
-      {({ value: pm }) => <div>{paymentMethodLabelWithIcon(intl, pm)}</div>}
-    </StyledSelect>
-  );
-};
+class PaymentMethodSelect extends Component {
+  static propTypes = {
+    onChange: PropTypes.func.isRequired,
+    /** To control the component state */
+    value: PropTypes.object,
+    /** The payment methods to display. **Cannot be empty !** */
+    paymentMethods: PropTypes.arrayOf(PropTypes.object).isRequired,
+    /** The default payment method. Will use the first one if not provided. */
+    defaultPaymentMethod: PropTypes.object,
+    /** @ignore Provided by injectIntl */
+    intl: PropTypes.object,
+  };
 
-PaymentMethodSelect.propTypes = {
-  /** The payment methods to display. **Cannot be empty !** */
-  paymentMethods: PropTypes.arrayOf(PropTypes.object).isRequired,
-  /** The default payment method. Will use the first one if not provided. */
-  defaultPaymentMethod: PropTypes.object,
-  /** @ignore Provided by injectIntl */
-  intl: PropTypes.object,
-};
+  paymentMethodValueAndLabel = (intl, paymentMethod) => {
+    return {
+      value: paymentMethod.id,
```

Comment:
If value is now an `id` (and not a `PaymentMethod`) then we need to update https://github.com/opencollective/opencollective-frontend/blob/93c482d57dc7410dcc0d03a635f37dcaee4e6632/components/CreateVirtualCardsForm.js#L162 as well, otherwise the select doesn't work

Reply:
I changed the value from id to payment method. the payment method select returns the payment method on change